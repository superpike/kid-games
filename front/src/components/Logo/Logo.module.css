.wrapperDark,
.wrapperLight {
    font-size: 20px;
    font-weight: bold;
}

.wrapperDark {
    color: #07071f;
}

.wrapperLight {
    color: #1c1c88;
}