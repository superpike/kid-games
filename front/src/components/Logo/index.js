import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import { withTheme } from '../../HOC/withTheme';

import styled from 'styled-components';

const Wrapper = styled.div`
  font-size: 20px;
  font-weight: bold;
  color: ${props => (props.theme === 'dark' ? '#07071f' : '#1c1c88')};
`;

export const Logo = ({ theme }) => {
  const { t } = useTranslation();
  return <Wrapper theme={theme}>{t('logoName')}</Wrapper>;
};

Logo.propTypes = {
  theme: PropTypes.oneOf(['dark', 'light']).isRequired,
};

export default withTheme(Logo);
