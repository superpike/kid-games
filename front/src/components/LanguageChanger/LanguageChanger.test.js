import React from 'react';
import { render } from '@testing-library/react';
import { LanguageChanger } from './index';
import { Logo } from '../Logo';
import '@testing-library/jest-dom/extend-expect';

describe('Language changer', () => {
  test('should be default language', () => {
    const { container } = render(
      <LanguageChanger defaultLanguage={'en'} />
    );
    expect(container).toMatchSnapshot();
  });
  test('should be list of languages', () => {
    const { container, getByText } = render(
      <LanguageChanger defaultLanguage={'en'} />
    );
    const button = container.querySelector('.fa-chevron-down')
      .parentElement;
    button.click();
    expect(getByText('en')).toBeInTheDocument();
    expect(getByText('ru')).toBeInTheDocument();
    expect(getByText('cz')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('should change language to ru', () => {
    const { container, getByText } = render(
      <LanguageChanger defaultLanguage={'en'} />
    );
    const { getByText: enGBT } = render(<Logo />);
    const button = container.querySelector('.fa-chevron-down')
      .parentElement;
    button.click();
    const elementForClick = getByText('ru');
    elementForClick.click();
    const { getByText: ruGBT } = render(<Logo />);
    expect(enGBT('Kid games')).toBeInTheDocument();
    expect(ruGBT('Детские игры')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
