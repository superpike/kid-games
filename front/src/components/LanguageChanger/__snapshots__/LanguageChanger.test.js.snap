// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language changer should be default language 1`] = `<div />`;
